reads_filtration/module_wraper.py

```diff
@@ -9,82 +11,173 @@
         return
 
-    os.makedirs(first_phase_output_path, exist_ok=True)
-    os.makedirs(logs_dir, exist_ok=True)
     if os.path.exists(first_phase_done_path):
         logger.info(f'{datetime.datetime.now()}: skipping reads_filtration step ({first_phase_done_path} already exists)')
-        return
+        return        
+        
+    os.makedirs(first_phase_output_path, exist_ok=True)
+    os.makedirs(logs_dir, exist_ok=True)
+        
+    barcode2samplename_dict = load_table_to_dict(barcode2samplename, 'Barcode {} belongs to more than one sample_name!!')
+    sample_names = sorted(barcode2samplename_dict.values())
+    error_path = error_path if error_path else os.path.join(first_phase_output_path, 'error.txt')
 
-    done_path = f'{logs_dir}/done_demultiplexing.txt'
-    logger.info('_' * 100)
-    logger.info(f'{datetime.datetime.now()}: demultiplexig sequences for {first_phase_output_path}')
+    script_name = 'filter_reads.py'
+    num_of_expected_results = 0
+    done_path = f'{logs_dir}/{exp_name}_done_demultiplexing.txt'
     if not os.path.exists(done_path):
-        # run filter_reads.py
-        parameters = [fastq_path, first_phase_output_path, logs_dir,
-                      done_path, barcode2samplename,
-                      f'--error_path {error_path}',
-                      f'--left_construct {left_construct}',
-                      f'--right_construct {right_construct}',
-                      f'--max_mismatches_allowed {max_mismatches_allowed}',
-                      f'--min_sequencing_quality {min_sequencing_quality}',
-                      f'--minimal_length_required {minimal_length_required}'] + (['--gz'] if gz else [])
-
-        fetch_cmd(f'{src_dir}/reads_filtration/filter_reads.py',
-                  parameters, verbose, error_path)
-        num_of_expected_results = 1
-        wait_for_results('filter_reads.py', logs_dir, num_of_expected_results,
-                         error_file_path=error_path, suffix='demultiplexing.txt')
+        logger.info('_' * 100)
+        logger.info(f'{datetime.datetime.now()}: demultiplexig sequences for {done_path}')
+        all_cmds_params = []
+        if not os.path.exists(done_path):
+            cmds = [fastq_path, first_phase_output_path, logs_dir,
+                    done_path, barcode2samplename, 'summary_log.txt',
+                    f'--error_path {error_path}',
+                    f'--left_construct {left_construct}',
+                    f'--right_construct {right_construct}',
+                    f'--max_mismatches_allowed {max_mismatches_allowed}',
+                    f'--min_sequencing_quality {min_sequencing_quality}',
+                    f'--minimal_length_required {minimal_length_required}'] + (['--gz'] if gz else [])
+            all_cmds_params.append(cmds)
+        else:
+            logger.debug(f'skipping filter reads as {done_path} found')
+            num_of_expected_results += 1
+        
+        if len(all_cmds_params)>0:
```

Comment:
When do we get more than one item in list?
The code is currently built to run each read sequentially,
For sequential run list is not needed (see original code).
You may want to refactor the code to support parallel reads